packages/store/addon/-private/system/store.ts

```diff
@@ -868,64 +950,91 @@ class Store extends Service {
     return Promise.all(fetches);
   }
 
-  _scheduleFetch(internalModel: InternalModel, options): Promise<InternalModel> {
-    if (internalModel._promiseProxy) {
-      return internalModel._promiseProxy;
-    }
+  _scheduleFetchThroughFetchManager(internalModel: InternalModel, options = {}): RSVP.Promise<InternalModel> {
+    let generateStackTrace = this.generateStackTracesForTrackedRequests;
+    let promise = this._fetchManager.scheduleFetch(identifierForIM(internalModel), options, generateStackTrace);
+    return promise.then(
+      payload => {
+        this._push(payload);
+        // Returning this._push here, breaks typing but not any tests, invesstigate potential missing tests
```

Comment:
we just need to type it in a way we can narrow as _find in finders prevents this being anything but an `SingleResourceDocument` which means the return will always be an `InternalModel`. See `push()` for typing example.

Reply:
Will do in a followup PR